lib/datasource/rubygems/get.ts

```diff
@@ -29,12 +33,83 @@ export async function fetch(
   return response.body;
 }
 
+export async function fetchDependencies(
+  dependencies: string[],
+  registry: string
+): Promise<MarshalledVersionInfo[]> {
+  const headers = getHeaders();
+
+  const url = urlJoin(
+    registry,
+    DEPENDENCIES_PATH,
+    '?' +
+      getQueryString({
+        gems: dependencies.join(','),
+      })
+  );
+
+  logger.trace({ dependencies }, `RubyGems lookup request: ${String(url)}`);
+  const response = await http.getBuffer(url, { headers });
+
+  logger.debug({ response }, 'dependencies response');
+
+  return new Marshal(response.body).parsed as MarshalledVersionInfo[];
+}
+
+async function getDependencyFromV1Dependencies(
+  dependency: string,
+  registry: string
+): Promise<ReleaseResult | null> {
+  logger.debug({ dependency }, 'falling back to dependencies endpoint');
+  let depInfo: MarshalledVersionInfo[] | null = null;
+  try {
+    depInfo = await fetchDependencies([dependency], registry);
+  } catch (err) {
+    // Ignore error because this endpoint is just a fallback
+    logger.debug({ registry, err }, 'dependencies endpoint returns error');
+  }
+  if (!depInfo) {
+    return null;
+  }
+  const releases = depInfo.map(
+    ({ number: version, platform: rubyPlatform }) => ({
+      version,
+      rubyPlatform,
+    })
+  );
+  return {
+    releases,
+    homepage: null,
+    sourceUrl: null,
+    changelogUrl: null,
+  };
+}
+
 export async function getDependency(
   dependency: string,
   registry: string
 ): Promise<ReleaseResult | null> {
   logger.debug({ dependency }, 'RubyGems lookup for dependency');
-  const info = await fetch(dependency, registry, INFO_PATH);
+  let info = null;
+  try {
+    info = await fetch(dependency, registry, INFO_PATH);
```

Comment:
If we know github packages has only the `/api/v1/dependencies` endpoint, we should directly use that to reduce calls.

Reply:
Do you mean something like `host === "rubygems.pkg.github.com"`?

Reply:
yes

Reply:
Done in 3039b09. Removed fallbacks instead as I think we only need to support GitHub Packages for now.